### PR TITLE
Skip armv7 for iOS11+ AppCheck

### LINKED ISSUE #7981: Nightly Testing Report
# Nightly Testing Report
This issue([log](https://github.com/firebase/firebase-ios-sdk/actions/runs/789174201)) is generated at 04/27/2021 06:09 AM, fetching workflow runs triggered in the last 24 hrs.
| Workflow |04/26|
| -------- | -------- |
| [zip](https://github.com/firebase/firebase-ios-sdk/blob/master/.github/workflows/zip.yml) | [failure](https://github.com/firebase/firebase-ios-sdk/actions/runs/788339791) |

### PR DESCRIPTION
**Pull request by paulb777**
Fix #7981

Surgical fix for building AppCheck and avoiding failure trying to build armv7. 

Makes the build order deterministic to avoid hard to repro issues and because AppCheck needs to be built last.

We'll need a more robust fix if we continue iOS 10 support for much longer.

#no-changelog

### CODE REVIEW

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift [hunk 1]
```diff
@@ -184,7 +184,18 @@ struct FrameworkBuilder {
...
+
+    var archs = targetPlatform.archs.map { $0.rawValue }.joined(separator: " ")
+    // The 32 bit archs do not build for iOS 11.
+    // TODO: Make a more robust solution if we need to support more of a mix between iOS 11 and
+    // under.
+    if framework == "FirebaseAppCheck" {
+      if targetPlatform == .iOSDevice {
+        archs = "arm64"
+      } else if targetPlatform == .iOSSimulator {
+        archs = "x86_64"
```
Comment (ryanwilson) [inline ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:196]: Could this also be `arm64` if it's on an ARM Mac?

**Comment by ryanwilson** [comment]
LGTM mostly, one question about Apple silicon Macs, plus reverting the `.yml` files afterwards.

**Review by ryanwilson — APPROVED** [review]

**Comment by paulb777** [comment]
The GHA failure is weird. It doesn't repro in a small example. I got it to reproduce in the debugger. Somehow a simple archive is being built instead of an architecture?

Compiling PromisesObjC for iphonesimulator (i386 x86_64 arm64) with command:
/usr/bin/xcodebuild build -configuration release -workspace /var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/FrameworkMaker.xcworkspace -scheme PromisesObjC GCC_GENERATE_DEBUGGING_SYMBOLS=NO ARCHS=i386 x86_64 arm64 VALID_ARCHS**=i386 x86_64 arm64** ONLY_ACTIVE_ARCH=NO BUILD_LIBRARY_FOR_DISTRIBUTION=YES SUPPORTS_MACCATALYST=NO BUILD_DIR=/var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/iphonesimulator -sdk iphonesimulator OTHER_CFLAGS=$(value) -DFIREBASE_BUILD_ZIP_FILE
2021-04-27 15:51:44.259833-0700 xcodebuild[19796:32669959] Metal API Validation Enabled
2021-04-27 15:51:44.455908-0700 xcodebuild[19796:32669889] warning:  The file reference for "GoogleAppMeasurement-xcframeworks.sh" is a member of multiple groups ("Support Files" and "Support Files"); this indicates a malformed project.  Only the membership in one of the groups will be preserved (but membership in targets will be unaffected).  If you want a reference to the same file in more than one group, please add another reference to the same path.
2021-04-27 15:51:44.455965-0700 xcodebuild[19796:32669889] warning:  The file reference for "GoogleAppMeasurement.debug.xcconfig" is a member of multiple groups ("Support Files" and "Support Files"); this indicates a malformed project.  Only the membership in one of the groups will be preserved (but membership in targets will be unaffected).  If you want a reference to the same file in more than one group, please add another reference to the same path.
2021-04-27 15:51:44.455991-0700 xcodebuild[19796:32669889] warning:  The file reference for "GoogleAppMeasurement.release.xcconfig" is a member of multiple groups ("Support Files" and "Support Files"); this indicates a malformed project.  Only the membership in one of the groups will be preserved (but membership in targets will be unaffected).  If you want a reference to the same file in more than one group, please add another reference to the same path.
Successfully built PromisesObjC for iphonesimulator. Build log is at file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/build_logs/PromisesObjC-iphonesimulator.txt.
buildDir: file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/iphonesimulator/
buildSliced returns file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/iphonesimulator/Release-iphonesimulator/PromisesObjC/FBLPromises.framework/
Compiling PromisesObjC for catalyst (x86_64 arm64) with command:
/usr/bin/xcodebuild build -configuration release -workspace /var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/FrameworkMaker.xcworkspace -scheme PromisesObjC GCC_GENERATE_DEBUGGING_SYMBOLS=NO ARCHS=x86_64 arm64 VALID_ARCHS=x86_64 arm64 ONLY_ACTIVE_ARCH=NO BUILD_LIBRARY_FOR_DISTRIBUTION=YES SUPPORTS_MACCATALYST=YES BUILD_DIR=/var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/catalyst -sdk macosx OTHER_CFLAGS=$(value) -DFIREBASE_BUILD_ZIP_FILE CODE_SIGN_IDENTITY=-
2021-04-27 15:51:48.889876-0700 xcodebuild[19827:32670303] Metal API Validation Enabled
2021-04-27 15:51:49.024656-0700 xcodebuild[19827:32670234] warning:  The file reference for "GoogleAppMeasurement-xcframeworks.sh" is a member of multiple groups ("Support Files" and "Support Files"); this indicates a malformed project.  Only the membership in one of the groups will be preserved (but membership in targets will be unaffected).  If you want a reference to the same file in more than one group, please add another reference to the same path.
2021-04-27 15:51:49.024701-0700 xcodebuild[19827:32670234] warning:  The file reference for "GoogleAppMeasurement.debug.xcconfig" is a member of multiple groups ("Support Files" and "Support Files"); this indicates a malformed project.  Only the membership in one of the groups will be preserved (but membership in targets will be unaffected).  If you want a reference to the same file in more than one group, please add another reference to the same path.
2021-04-27 15:51:49.024725-0700 xcodebuild[19827:32670234] warning:  The file reference for "GoogleAppMeasurement.release.xcconfig" is a member of multiple groups ("Support Files" and "Support Files"); this indicates a malformed project.  Only the membership in one of the groups will be preserved (but membership in targets will be unaffected).  If you want a reference to the same file in more than one group, please add another reference to the same path.
Successfully built PromisesObjC for catalyst. Build log is at file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/build_logs/PromisesObjC-catalyst.txt.
buildDir: file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/catalyst/
buildSliced returns file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/catalyst/Release-maccatalyst/PromisesObjC/FBLPromises.framework/
fatal error: /Applications/Xcode12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: input file (/var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/iphonesimulator/Release-iphonesimulator/PromisesObjC/FBLPromises.framework/FBLPromises) must be a fat file when the -thin option is specified
Fatal error: lipo command exited with 1 when trying to extract the i386 slice from /var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/iphonesimulator/Release-iphonesimulator/PromisesObjC/FBLPromises.framework/FBLPromises. Output:
The task completed: file zip_builder/FrameworkBuilder.swift, line 850
2021-04-27 15:51:52.476623-0700 zip-builder[93903:32580995] Fatal error: lipo command exited with 1 when trying to extract the i386 slice from /var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/iphonesimulator/Release-iphonesimulator/PromisesObjC/FBLPromises.framework/FBLPromises. Output:
The task completed: file zip_builder/FrameworkBuilder.swift, line 850
(lldb) po binary
▿ file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/iphonesimulator/Release-iphonesimulator/PromisesObjC/FBLPromises.framework/FBLPromises
  - _url : file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/iphonesimulator/Release-iphonesimulator/PromisesObjC/FBLPromises.framework/FBLPromises

(lldb) po frameworks
▿ 3 elements
  ▿ 0 : 2 elements
    - key : zip_builder.TargetPlatform.iOSSimulator
    ▿ value : file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/iphonesimulator/Release-iphonesimulator/PromisesObjC/FBLPromises.framework/
      - _url : file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/iphonesimulator/Release-iphonesimulator/PromisesObjC/FBLPromises.framework/
  ▿ 1 : 2 elements
    - key : zip_builder.TargetPlatform.iOSDevice
    ▿ value : file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/iphoneos/Release-iphoneos/PromisesObjC/FBLPromises.framework/
      - _url : file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/iphoneos/Release-iphoneos/PromisesObjC/FBLPromises.framework/
  ▿ 2 : 2 elements
    - key : zip_builder.TargetPlatform.catalyst
    ▿ value : file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/catalyst/Release-maccatalyst/PromisesObjC/FBLPromises.framework/
      - _url : file:///var/folders/mk/9ccbvq8x00q20q0f7mfxbtrw00bzqz/T/ZipRelease/2021-04-27T15-27-11/project-ios/catalyst/Release-maccatalyst/PromisesObjC/FBLPromises.framework/

(lldb)

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift (hunk 1 above)
Comment (paulb777) [reply ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:196]: Good catch!

**Comment by paulb777** [comment]
Going to merge here since the nightly is broken anyway and this might work.

I'll address any review and follow-on issues in a subsequent PR

**Pull request merged**